lib/runtime/src/lib.rs

```diff
@@ -320,7 +327,19 @@ pub fn default_compiler() -> impl Compiler {
     #[cfg(any(feature = "default-backend-cranelift", feature = "docs"))]
     use wasmer_clif_backend::CraneliftCompiler as DefaultCompiler;
 
-    DefaultCompiler::new()
+    #[cfg(any(
+        feature = "default-backend-singlepass",
+        feature = "default-backend-cranelift",
+        feature = "default-backend-llvm",
+    ))]
+    return Box::new(DefaultCompiler::new());
+
+    #[cfg(not(any(
```

Comment:
I tried your suggestion but it turns out `DefaultCompiler` will be not defined if the return statement if doesn't have the `#[cfg(...)]` the line before, and causes Rust compilation to fail.

Reply:
I'm going to mark this as resolved, since there seem to be no obvious way to do it properly.

Perhaps we can open another PR apart using the `cfg-if` crate :)